packages/jest-circus/src/__tests__/tsconfig.json

```diff
@@ -3,5 +3,6 @@
   "compilerOptions": {
     "rootDir": "../"
   },
-  "include": ["../**/*"]
+  "include": ["../**/*"],
+  "references": [{"path": "../../../jest-expect"}]
```

Comment:
Needed to include this declaration:

https://github.com/facebook/jest/blob/6dec46ec864400554846180d55cc085813d1c232/packages/jest-expect/src/types.ts#L50-L57

Reply:
Shouldn't types be built before type testing tests?

Reply:
In this case `tsc` needs the reference for augmentation. The `packages/jest-circus/tsconfig.json` has this reference, but the `packages/jest-circus/src/__tests__/tscongo.json` is extending the root config. That’s why the reference is missing.

Reply:
Reference is only so the build is in the correct order right? I don't understand why the test builds needs a specific order

Reply:
I also thought so, but typecheck was not passing. Just removed `references` and now it is passing. Not sure why? ;D Let’s see what CI thinks.

Reply:
See the error in CI. With references it goes away. I did not know that, just guessed that reference might work. Adding a `.d.ts` also works, but reference seemed to be cleaner.